core/commands/IssueUpdateCommand.php

```diff
@@ -0,0 +1,621 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+require_api( 'access_api.php' );
+require_api( 'authentication_api.php' );
+require_api( 'bug_api.php' );
+require_api( 'config_api.php' );
+require_api( 'constant_inc.php' );
+require_api( 'custom_field_api.php' );
+require_api( 'date_api.php' );
+require_api( 'email_api.php' );
+require_api( 'error_api.php' );
+require_api( 'event_api.php' );
+require_api( 'file_api.php' );
+require_api( 'helper_api.php' );
+require_api( 'lang_api.php' );
+require_api( 'last_visited_api.php' );
+require_api( 'profile_api.php' );
+require_api( 'relationship_api.php' );
+require_api( 'string_api.php' );
+require_api( 'user_api.php' );
+
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_api.php' );
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_enum_api.php' );
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_issue_api.php' );
+require_once( dirname( __FILE__ ) . '/../../api/soap/mc_project_api.php' );
+
+use Mantis\Exceptions\ClientException;
+
+/**
+ * Sample:
+ * {
+ *   "query": {
+ *      "issue_id": 1234,
+ *   },
+ *   "payload": {
+ *      "issue": ... see rest issue update documentation
+ *      "notes": ... see rest issue update documentation
+ *   },
+ *   "options: {
+ *      "action_type": BUG_UPDATE_TYPE_NORMAL,
+ *      "return_issue": true,
+ *   }
+ * }
+ */
+
+/**
+ * A command that updates an issue.
+ */
+class IssueUpdateCommand extends Command {
+    /**
+	 * The existing issue.
+	 *
+	 * @var BugData
+	 */
+	private $existing_issue = null;
+
+	/**
+	 * The updated issue.
+	 *
+	 * @var BugData
+	 */
+	private $updated_issue = null;
+
+	/**
+	 * The existing bugnote.
+	 *
+	 * @var BugNoteData
+	 */
+	private $existing_notes = array();
+
+	/**
+	 * The updated bugnote.
+	 *
+	 * @var BugNoteData
+	 */
+	private $updated_notes = array();
+
+	/**
+	 * @var integer
+	 */
+	private $user_id;
+
+	/**
+	 * The custom fields to set.
+	 */
+	private $custom_fields_to_set = array();
+
+	/**
+	 * The resolve issue status.
+	 *
+	 * * @var bool
+	 */
+	private $resolve_issue = false;
+
+	/**
+	 * The close issue status.
+	 *
+	 * * @var bool
+	 */
+	private $close_issue = false;
+
+	/**
+	 * The reopen issue status.
+	 *
+	 * * @var bool
+	 */
+	private $reopen_issue = false;
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $p_data The command data.
+	 */
+	function __construct( array $p_data ) {
+		parent::__construct( $p_data );
+	}
+
+	/**
+	 * Validate the data.
+	 */
+	protected function validate() {
+		$this->user_id = auth_get_current_user_id();
+
+		$t_update_type	 = $this->option( 'action_type', BUG_UPDATE_TYPE_NORMAL );
+		$t_issue_id	 = helper_parse_issue_id( $this->query( 'issue_id' ) );
+
+		$t_update_data	 = $this->payload( 'issue' );
+		$t_notes	 = $this->payload( 'notes' );
+
+		if( !isset( $t_update_data['last_updated'] ) || is_blank( $t_update_data['last_updated'] ) ) {
```

Comment:
In the rest api, this field is called `updated_at`.   Update accordingly everywhere.  Make sure the same is the case for notes.

Reply:
In REST API, we use etags rather than a `updated_at`.  See the current implementation in `issues_rest.php`.  Hence, this field shouldn't be allowed in the payload.  If needed for internal calls, it should be done via options array.